f/frizzle/comapeo/comapeo_observations.py

```diff
@@ -92,20 +108,50 @@ def fetch_comapeo_projects(
             if project["project_id"] not in comapeo_project_blocklist
         ]
 
-    if not comapeo_projects:
-        raise ValueError("No valid projects found.")
-
     return comapeo_projects
 
 
+@retry(
+    stop_max_attempt_number=3,
+    wait_exponential_multiplier=500,
+)
+def download_attachment(url, headers, save_path):
+    try:
+        response = requests.get(url, headers=headers)
+
+        content_type = response.headers.get("Content-Type", "")
+        extension = mimetypes.guess_extension(content_type) or ""
+
+        file_name = os.path.basename(url) + extension
+        os.makedirs(os.path.dirname(save_path), exist_ok=True)
+        with open(save_path, "wb") as file:
+            file.write(response.content)
```

Comment:
```suggestion
        with open(save_path, "wb") as f:
            f.write(response.content)
```

`file` is a reserved word in Python